### PR TITLE
Update device flow OAuth app responses

### LINKED ISSUE #3944: Device flow: clarify that responses are based on "Accept" header
### What article on docs.github.com is affected?

https://docs.github.com/en/developers/apps/authorizing-oauth-apps

### What part(s) of the article would you like to see updated?

1. [`POST https://github.com/login/device/code` response section](https://docs.github.com/en/developers/apps/authorizing-oauth-apps#response-1)
2. [`POST https://github.com/login/oauth/access_token` response section](https://docs.github.com/en/developers/apps/authorizing-oauth-apps#response-2)

### Additional information

By default, all responses to `POST https://github.com/login/*` are a form query, as it is already documented for the [`POST https://github.com/login/oauth/access_token ` response](https://docs.github.com/en/developers/apps/authorizing-oauth-apps#response) of the OAuth web flow. All endpoints optionally accept the `Accept` header to be set to `application/json` or `application/xml` to alter the response type. The two sections for the device flow need to be updated to reflect that. They currently imply that the response is JSON by default, which is incorrect

### PR DESCRIPTION
**Pull request by lucascosti**
### Why: OAuth device flow: clarify that responses are based on "Accept" header

Closes https://github.com/github/docs/issues/3944

### What's being changed:

As suggested by the awesome @gr2m, I've clarified the responses for the device flow requests, and made it consistent with the style in the web flow one.

I also tested the flow out myself, made the responses consistent, and updated the screenshot for the device code form.

Stage: https://gha-docs-10426--lucascostioaut.herokuapp.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps

### Check off the following:

- [ ] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### CODE REVIEW

File: data/reusables/apps/oauth-auth-vary-response.md [hunk 1]
```diff
@@ -0,0 +1 @@
+You can also receive the content in different formats if you provide the format in the `Accept` header. For example, `Accept: application/json` or `Accept: application/xml`:
```
Comment (lucascosti) [inline data/reusables/apps/oauth-auth-vary-response.md:1]:
```suggestion
You can also receive the response in different formats if you provide the format in the `Accept` header. For example, `Accept: application/json` or `Accept: application/xml`:
```

**Review by gr2m — APPROVED** [review]
Looks good to me 👍🏼

**Review by ethanpalm — APPROVED** [review]
These changes look good to go 🎉

**Comment by GrandSlam4k** [comment]
@Gramzman28

**Pull request merged**